astropy/table/tests/test_table.py

```diff
@@ -2456,6 +2456,21 @@ class MyTable(table.Table):
         assert type(t['e']) is MySubMaskedColumn  # sub-class not downgraded
 
 
+def test_sort_with_non_mutable():
+    """Test sorting a table that has a non-mutable column such as SkyCoord"""
+    t = Table([[2, 1], SkyCoord([4, 3], [6, 5], unit='deg,deg')], names=['a', 'sc'])
+    meta = {'a': [1, 2]}
+    t['sc'].info.meta = meta
+    t.sort('a')
+    assert np.all(t['a'] == [1, 2])
+    assert np.allclose(t['sc'].ra.to_value(u.deg), [3, 4])
+    assert np.allclose(t['sc'].dec.to_value(u.deg), [5, 6])
+
+    # Got a deep copy of SkyCoord column
```

Comment:
Are you sure you want to test this? This would break if eventually we change it so sorting can be done in-place. My sense is that all we want to test is that the `meta` are the same. Or were you thinking that the eventual goal should be anyway not to have this inplace, so this is desired behaviour that should stay this way? 

Thinking about this, I'm happy either way, but maybe make it explicit in the comment.

Reply:
I am happy enough if tests break later since that gives a clear indication of a behavior change, and then we understand what's going on and update the test.  These sort of tests were invaluable in all the monkeying around I did with meta copying.

Reply:
OK, makes sense.